src/operations/grants.ts

```diff
@@ -0,0 +1,65 @@
+import { MigrationOptions } from '../types'
+import { GrantRoles, GrantOnTables, GrantOnSchemas, GrantOnTablesProps, GrantOnAllTablesProps } from './grantsTypes'
+
+export { GrantRoles, GrantOnTables, GrantOnSchemas }
+
+const isArray = <T>(item: T | T[]): item is T[] => {
+  return typeof item !== 'string' && Boolean((item as Array<T>).length !== undefined)
+}
+
+const asArray = <T>(item: T | T[]) => (isArray(item) ? item : [item])
+
+const isGrantOnAllTablesProps = (props: GrantOnTablesProps): props is GrantOnAllTablesProps => {
+  return 'schema' in props
+}
+
+export function grantRoles(mOptions: MigrationOptions) {
+  const _grantRoles: GrantRoles = (rolesFrom, rolesTo) => {
+    const rolesFromStr = asArray(rolesFrom).map(mOptions.literal).join(',')
+    const rolesToStr = asArray(rolesTo).map(mOptions.literal).join(',')
+    return `GRANT ${rolesFromStr} TO ${rolesToStr};`
+  }
+  _grantRoles.reverse = () => {
+    console.log('reverse')
+    return ''
+  }
+  return _grantRoles
+}
+
+export function grantOnTables(mOptions: MigrationOptions) {
+  const _grantOnTables: GrantOnTables = (props) => {
+    const { privileges, roles, withGrantOption } = props
+    const rolesStr = asArray(roles).map(mOptions.literal).join(',')
+    const privilegesStr = asArray(privileges).map(String).join(',')
+    let tablesStr
```

Comment:
It's usually a good idea to type `let`s that has no initial assigned values
In this case it should be 

`let tablesStr: string` or `let tablesStr: string | undefined`